charts/airflow/templates/_helpers/common.tpl

```diff
@@ -100,6 +100,23 @@ true
 {{- end -}}
 {{- end -}}
 
+{{/*
+If the airflow dag processor should be used.
+*/}}
+{{- define "airflow.dagProcessor.should_use" -}}
+{{- if .Values.dagProcessor.enabled -}}
+{{- if not .Values.airflow.legacyCommands -}}
+{{- if include "airflow.image.version" . -}}
+{{- if semverCompare ">=2.3.0" (include "airflow.image.version" .) -}}
+true
+{{- end -}}
+{{- else -}}
+false
```

Comment:
The truthiness of this `airflow.dagProcessor.should_use` is actually simply whether it is defined.

In this case, we actually do want to allow the dag processor (if we don't know what version the user is using). Because in that case, they are probably using a custom image. All this is to say, we should make this string say `true`.